proposals/jaeger.adoc

```diff
@@ -0,0 +1,169 @@
+* Use cpf13/cobra for flexible configuration under Kubernetes - https://github.com/uber/jaeger/issues/233
+* Instrumentation libraries in more languages - https://github.com/uber/jaeger/issues/366
+
+*Social media accounts:*
+
+* Blog: https://medium.com/jaegertracing/
+* Twitter: https://twitter.com/JaegerTracing
+
+*External Dependencies:*
+
+* https://github.com/uber/jaeger
+** https://github.com/pkg/errors - BSD-2
+** https://github.com/uber-go/zap - MIT
+** https://github.com/apache/thrift - AL2
+** https://github.com/opentracing/opentracing-go - MIT
+** golang.org/x/net - BSD-3
+** https://github.com/uber/tchannel-go - MIT
+** https://github.com/gorilla/ - BSD-3
+** https://github.com/kr/pretty - MIT
+** https://github.com/go-kit/kit - MIT
+** http://github.com/olivere/elastic - MIT
+* https://github.com/jaegertracing
+** Apache-2.0
+* https://github.com/uber/jaeger-lib
+** https://github.com/codahale/hdrhistogram - MIT
+** https://github.com/go-kit/kit - MIT
+** https://github.com/uber-go/tally - MIT
+** https://github.com/prometheus/client_golang - AL2
+* https://github.com/uber/jaeger-ui
+** Many (see full list - https://gist.github.com/yurishkuro/273b5c0ed3556dfb84a095002f7dc40c)
+** "(GPL-2.0 OR MIT)"
+** "Apache-2.0"
+** "BSD"
+** "BSD*"
+** "BSD-2-Clause"
+** "BSD-3-Clause"
+** "ISC"
+** "MIT"
+** "MIT*"
+* https://github.com/uber/jaeger-client-go
+** https://github.com/apache/thrift - AL2
+** https://github.com/opentracing/opentracing-go - MIT
+** golang.org/x/net - BSD-3
+** https://github.com/uber/tchannel-go - MIT
+* https://github.com/uber/jaeger-client-java
+** https://mvnrepository.com/artifact/org.apache.thrift/libthrift - AL2
+** https://mvnrepository.com/artifact/io.opentracing/opentracing-api - AL2
+** https://mvnrepository.com/artifact/com.google.code.gson/gson - AL2
+** https://mvnrepository.com/artifact/org.slf4j/slf4j-api - MIT
+* https://github.com/uber/jaeger-client-node
+** https://www.npmjs.com/package/node-int64 - MIT
+** https://www.npmjs.com/package/thriftrw - MIT
+** https://www.npmjs.com/package/xorshift - MIT
+** https://www.npmjs.com/package/opentracing - MIT
+* https://github.com/uber/jaeger-client-python
+** https://pypi.python.org/pypi/futures - PSF
+** https://pypi.python.org/pypi/threadloop - MIT
+** https://pypi.python.org/pypi/thrift - AL2
+** https://pypi.python.org/pypi/tornado - AL2
+** https://pypi.python.org/pypi/opentracing - MIT
+** https://pypi.python.org/pypi/future - MIT
+
+*Statement on alignment with CNCF mission:*
+
+A challenging part of cloud native computing is managing microservices and especially debugging the situation
+when things go awry. Furthermore, distributed tracing instrumentation has been fragmented traditionally until
+CNCF efforts like the OpenTracing specification came to existence to help unify existing tracing implementations
+out there.
+
+Jaeger is a battle tested distributing system that takes advantage of OpenTracing and advances the state
```

Comment:
Can provide more detail on what you mean by battle tested?  At what level of scale has this tracing been tested and what are the performance implications?   In other open source cloud  infrastructure communities a monitoring project was accepted for inclusion before it could prove it could scale and before it could prove that it could work for both private and public clouds.   This was a huge mistake as that monitoring project's architecture prohibited it from working at very large scale and the project was accepted accepted yet never able to meet the requirements of the overall community.   For a project like this I would really recommend that you keep the bar very high and make sure the community needs are well satisfied before a project like this goes beyond incubator status.

Reply:
Regarding "battle tested", Jaeger has been running in production at Uber for two years, as of today tracing more than 1200 microservices. We're using fairly aggressive sampling rates (typically between 1/1000 and 1/10000 for top tier services), which make performance overhead negligible. Based on anecdotal evidence from other commercial vendors, tracing 100% of traffic may incur up to 5% of performance overhead, which can be tuned by less extensive instrumentation and further perf optimization in the tracing libraries.

Regarding the scalability of Jaeger architecture, all layers in Jaeger backend are horizontally scalable. Typically the bottleneck is in the storage backends (Cassandra, ES), which are also horizontally scalable. There are may avenues for improving the storage backend throughput and indexing capabilities, as well as reducing the amount of data needed to be stored in the first place, by investing in techniques likes on-demand sampling, post-trace sampling, trace discovery through aggregates rather than heavy indexing.

Reply:
1200 services does that mean 1200 different software projects, or 1200 running daemons (possibly multiple copies of the same code). though does it matter? can you shed some light on the volume of spans and traces being generated, or the volume of log lines, tags, etc, whatever it is that affects workload.

> anecdotal evidence from other commercial vendors, tracing 100% of traffic may incur up to 5% of performance overhead

can you clarify whether this means commercial vendors selling jaeger-based solutions, or alternatives to jaeger? if so, do we have stats on the overhead of jaeger?

Reply:
1200 different services / applications / products, not just different instances - those are probably in the 500,000 ballpark. Our tracing backend receives ~50K spans per second, median span size is ~500b, max ~60Kb.

> can you clarify whether this means commercial vendors selling jaeger-based solutions, or alternatives to jaeger? if so, do we have stats on the overhead of jaeger?

I meant commercial alternatives to Jaeger, like Lightstep, but my point was that the actual instrumentation libraries are very similar in functionality / architecture, so I am fairly comfortable extrapolating their perf overhead numbers. We don't have stats on perf overhead of Jaeger libs at 100% sampling - it's actually a non-linear problem with many contributing factors, so raw numbers like 5% are not very meaningful without describing the exact methodology and conditions of measurements. I guess what I'm getting at is that it's always possible to limit perf overhead by applying sampling, and even with low sampling rates like ours we get extremely rich data about application behavior that the overhead is totally worth it.